### PR TITLE
 [release-4.6] Use index image for the dev flow

### PR DESCRIPTION
**Pull request by yanirq**
Cherry-pick **Use index image for the dev flow**  from the following PRs:
https://github.com/openshift-kni/performance-addon-operators/pull/410
https://github.com/openshift-kni/performance-addon-operators/pull/427
https://github.com/openshift-kni/performance-addon-operators/pull/428
https://github.com/openshift-kni/performance-addon-operators/pull/429

### CODE REVIEW

**Review by fromanirh — COMMENTED** [review]
/lgtm
/hold

File: hack/common.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+VENDOR_DIR="${REPO_DIR}/vendor"
+OUT_DIR="${REPO_DIR}/build/_output"
+OUT_BIN="${OUT_DIR}/bin"
+METADATA_DIR="${REPO_DIR}/deploy/metadata/performance-addon-operator"
+
+IMAGE_BUILD_CMD=${IMAGE_BUILD_CMD:-podman}
+
+CSV_VERSION=${CSV_VERSION:-"4.6.0"}
+PREV_CSV_VERSION=${PREV_CSV_VERSION:-"4.5.0"}
+CSV_CHANNEL=${CSV_CHANNEL:-"4.6"}
```
Comment (cynepco3hahue) [inline hack/common.sh:16]: nit: new line

File: hack/generate-index-database.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/bin/sh -c "opm index add --mode semver --bundles ${BUNDLES} --generate")
+
+${IMAGE_BUILD_CMD} wait "${opm_container_id}"
+${IMAGE_BUILD_CMD} logs "${opm_container_id}"
+
+rm -rf "${OUT_DIR}/index.Dockerfile"
+rm -rf "${OUT_DIR}/database"
+
+${IMAGE_BUILD_CMD} cp "${opm_container_id}:/index.Dockerfile" "${OUT_DIR}"
+${IMAGE_BUILD_CMD} cp "${opm_container_id}:/database" "${OUT_DIR}"
```
Comment (cynepco3hahue) [inline hack/generate-index-database.sh:32]: nit: new line

**Pull request merged**